src/main/java/dev/snowdrop/release/services/AutoUpdateService.java

```diff
@@ -0,0 +1,165 @@
+            variableMap.put("majorVersion", releaseMMFA[0]);
+            variableMap.put("minorVersion", releaseMMFA[1]);
+            variableMap.put("patchVersion", releaseMMFA[2]);
+            variableMap.put("qualifier", qualifier);
+            variableMap.put("milestone", milestone);
+            BuildConfig buildConfigObj = factory.createFromRepo(buildConfigFile);
+            final String[] releaseMMF = releaseVersion.split("\\.");
+            final String releaseMM = releaseMMF[0] + "." + releaseMMF[1];
+            release.getComponents().stream().forEach(component -> {
+//            // TODO: Check if it's a product or a component only template
+//            manageComponentOnly(component);
+                manageProduct(component, variableMap);
+            });
+
+            factory.saveTo(buildConfigObj, generateVariableConfiguration(variableMap), buildConfigFile);
+            return buildConfigFile;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String generateVariableConfiguration(Map<String, String> variableMap) {
+        return variableMap.entrySet()
+            .stream()
+            .map(entry -> "#!" + entry.getKey() + "=" + entry.getValue())
+            .collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    private void manageProduct(Component component, Map<String, String> variableMap) {
+        final Issue issue = issueSvc.getIssue(component);
+        final String description = issue.getDescription();
+        final String versionSection = description.substring(description.lastIndexOf("==="));
+        final String[] artifactArr = versionSection.split("\\n[\\n \\r]*");
+        // Skip the 1st 2 lines
+        for (int arrayPos = 0; arrayPos < ((artifactArr.length - 1) / 5); arrayPos++) {
+            final String gavText = artifactArr[arrayPos * 5 + 1];
+            if (GAV_NAME_REGEX_PATTERN.reset(gavText).matches()) {
+                final String gid = GAV_NAME_REGEX_PATTERN.group(2);
+                final String aid = GAV_NAME_REGEX_PATTERN.group(4);
+                final String varName = WordUtils.capitalizeFully(aid, '-', '.').replaceAll("-", "");
+                final String productNameText = artifactArr[arrayPos * 5 + 1 + 1];
+                final String productVersionText = artifactArr[arrayPos * 5 + 1 + 2];
+                final String productVersion = productVersionText.split(":")[1];
+                final String supportedVersionText = artifactArr[arrayPos * 5 + 1 + 3];
+                final String supportedVersion = supportedVersionText.split(":")[1];
+                final String eolText = artifactArr[arrayPos * 5 + 1 + 4];
+                final var gavProd = toCamelCase(gid) + toCamelCase(aid) + "Prod";
+                if (variableMap.containsKey(gavProd)) {
+                    if (!supportedVersion.stripLeading().startsWith("[")) {
```

Comment:
Please explain why this is needed and why you strip `supportedVersion` twice. It'd be better to call `trim` once such as `supportedVersion = supportedVersion.trim()`.

Reply:
fixed.